src/platform/nodejs/persist/fileSystemPersist.ts

```diff
@@ -0,0 +1,132 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import { Logger } from '@opentelemetry/api';
+import { NoopLogger } from '@opentelemetry/core';
+import { PersistentStorage } from '../../../types';
+import { AzureExporterConfig, DEFAULT_EXPORTER_CONFIG } from '../../../config';
+import { confirmDirExists, getShallowDirectorySize } from './fileSystemHelpers';
+
+export class FileSystemPersist implements PersistentStorage {
+  static TEMPDIR_PREFIX = 'ot-azure-exporter-';
+
+  maxBytesOnDisk: number = 50_000_000; // ~50MB
+
+  private readonly _options: AzureExporterConfig;
+
+  private readonly _logger: Logger;
+
+  constructor(options: Partial<AzureExporterConfig> = {}) {
+    this._options = { ...DEFAULT_EXPORTER_CONFIG, ...options };
+    this._logger = options.logger || new NoopLogger();
+    if (!this._options.instrumentationKey) {
+      this._logger.error(
+        `No instrumentation key was provided to FileSystemPersister. Files may not be properly persisted`,
```

Comment:
Is there any way telemetry from a different iKey is saved here?, file name could be a problem is possible

Reply:
the directory is suffixed with ikey, so they will be separate